cms/navigation/models.py

```diff
@@ -0,0 +1,132 @@
+from typing import ClassVar
+
+from django.core.exceptions import ValidationError
+from django.db import models
+from wagtail.admin.panels import FieldPanel
+from wagtail.blocks import CharBlock, ListBlock, PageChooserBlock, StructBlock, URLBlock
+from wagtail.contrib.settings.models import BaseSiteSetting, register_setting
+from wagtail.snippets.models import register_snippet
+
+from cms.core.fields import StreamField
+
+
+# Custom LinkBlock to support both pages and URLs
+class BaseLinkBlock(StructBlock):
+    page = PageChooserBlock(required=False)
+    url = URLBlock(required=False, label="URL")
+    title = CharBlock(required=False, help_text="Optional. Displayed as the link text.")
+
+    def clean(self, value: dict) -> dict:
+        value = super().clean(value)
+        if not value.get("page") and not value.get("url"):
```

Comment:
Some validation is needed when there is both a url and a page added, as this should not be possible.

Reply:
Done!